.github/workflows/docs.yaml

```diff
@@ -12,7 +12,32 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  check_changes:
+    name: Deduce required tests from code changes
+    runs-on: ubuntu-latest
+    outputs:
+      docs-tree: ${{ steps.docs-tree.outputs.src }}
+    steps:
+      - name: Check code changes
+        uses: dorny/paths-filter@78ab00f87740f82aec8ed8826eb4c3c851044126
+        id: docs-tree
+        with:
+          base: ${{ github.event.pull_request.base.sha || github.event.before }}
+          filters: |
+            src:
+              - 'Documentation/**'
+              - 'bugtool/cmd/**'
+              - 'cilium/cmd/**'
+              - 'cilium-health/cmd/**'
+              - 'daemon/cmd/**'
+              - 'hubble-relay/cmd/**'
+              - 'operator/cmd/**'
```

Comment:
For workflows where the whole workflow should be skipped, and not only specific jobs, it would be more efficient to use the [`on.<push|pull_request>.paths`](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths) filters instead.

The `dorny/paths-filter` is nice because it allows fine-grained skipping at the job or steps-level, but it still requires a GH Action slot at least for the check, whereas the `on` filter does not.

Reply:
Oh, right! I guess I started with paths-filter and went full on :smile: I'll switch the smoke test, `kind-1.19.yaml`, and the docs test to use GitHub's path filters.

> it still requires a GH Action slot at least for the check, whereas the on filter does not.

Note that this doesn't matter much. Path filters take a slot for only around 3s so it's negligible compared to our other jobs lasting 10s of minutes.

Reply:
(Nitpicking: not really important but I think it's a bit more than 3s: the run itself might be 3s but before a run actually starts, the VM setup takes some time and reserves a slot, and this time is not displayed in the run time. With `on` filters, GitHub will directly skip the workflow and thus don't even add it to the queued runs, which saves a VM slot)